backend/schema/fsm.go

```diff
@@ -0,0 +1,109 @@
+package schema
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	schemapb "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/schema"
+	"github.com/TBD54566975/ftl/internal/slices"
+)
+
+type FSM struct {
```

Comment:
Since some portion of users might mistakenly think FSM stands for flying spaghetti monster instead of finite state machine, what would be the best place to potentially add documentation? Doc comments on the public API in go-runtime? I'll also need to add some docs on enum to the same place since the sumtype interface design could use a nice user guide. :)

Reply:
I suppose for that matter, maybe it's also worth adding a `// Finite state machine` in this file in case any of our future teammates get confused

Reply:
Ahhh perfect, just saw you added that in https://github.com/TBD54566975/ftl/pull/1412. That makes sense. I'll talk to Lizzy separately about where we should write a brief enum user guide